guides/source/getting_started.md

```diff
@@ -1991,6 +1991,140 @@ the `app/views/comments` directory.
+
+ You can use concerns in your controller or model the same way you would use any module. When you first created your app with `rails new blog` , two folders were created within `apps/` along with the rest:
+
+ ```
+ app/controllers/concerns
+ app/models/concerns
+ ```
+
+A given blog article might have various statuses - for instance, it may be `public` (visible to everyone), or `private` (only visible by the author, perhaps if an author wants to save a draft). It may also be `archived`, which means it is hidden from the author. A comment, similarly, may be `public`, `private`, or `archived`. For any given article, in the controller you might want to verify that an article is public before rendering it.
+
+Within the `article` model, after running a migration to add a `status` column, you might add:
+
+```ruby
+class Article < ApplicationRecord
+  has_many :comments
+  validates :title, presence: true,
+                    length: { minimum: 5 }
+
+  VALID_STATUSES = ['public', 'private', 'archived']
+
+  validates :status, in: VALID_STATUSES
+
+  def archived? 
+    status == 'archived'
+  end
+end
+```
+
+and in the `Comment` model:
+
+```ruby
+class Comment < ApplicationRecord
+  belongs_to :article
+
+  VALID_STATUSES = ['public', 'private', 'archived']
+
+  validates :status, in: VALID_STATUSES
+
+  def archived?
+    status == 'archived'
+  end
+end
+```
+
+Then, in our `index` action template (`app/views/articles/index.html.erb`) we would use the `archived?` method to avoid displaying any article that is archived:
+
+```html+erb
+<h1>Listing Articles</h1>
+<%= link_to 'New article', new_article_path %>
+<table>
+  <tr>
+    <th>Title</th>
+    <th>Text</th>
+    <th colspan="3"></th>
+  </tr>
+
+  <% @articles.each do |article| %>
+    <% unless article.archived? %> 
+      <tr>
+        <td><%= article.title %></td>
+        <td><%= article.text %></td>
+        <td><%= link_to 'Show', article_path(article) %></td>
+        <td><%= link_to 'Edit', edit_article_path(article) %></td>
+        <td><%= link_to 'Destroy', article_path(article),
+                method: :delete,
+                data: { confirm: 'Are you sure?' } %></td>
+      </tr>
+    <% end %>
+  <% end %>
+</table>
+```
+
+However, if you look again at our models now, you can see that the logic is duplicated. If in future we increase the functionality of our blog - to include private messages, for instance -  we might find ourselves duplicating the logic yet again. This is where concerns come in handy.
+
+Let's call our new concern (module) `Statusable`, a module to use for any model that has a status. We can create a new file inside `app/models/concerns` called `statuslike.rb` , and store all of the status methods that were duplicated in the models.
```

Comment:
`Statusable` and `Statuslike` are used inconsistently within these examples. I'd also suggest `Archivable` as a clearer name for this concern.

Reply:
Whoops! I intended to fully replace Statuslike with Statusable. Archivable is a pretty good name though! I'll probably go with that, though it might hide the fact that this also controls general visibility. Maybe `Viewable` , or is that too generic?

Reply:
`Viewable` does sound a tad generic to me. We could also reduce the scope of the example to only focus on archiving, after all it's more about showing how to use concerns than the specific functionality they introduce.

Reply:
In practice, I might add two concerns `Draftable` and `Archivable` to describe the public/private and published/archived functionality.

Reply:
@dzunk  Hmm, I can see where you're coming from, but I think that adding two concerns in this guide might be a bit confusing for a new reader. I think I might go along with one of the name examples in this [DHH](https://signalvnoise.com/posts/3372-put-chubby-models-on-a-diet-with-concerns) article - the `Visible` concern they mention. It's still somewhat generic but visible as a word most accurately reflects the intention behind the status (while actually being a proper word as well, unlike `Viewable`. ) What do you think?